src/users/users.service.ts

```diff
@@ -26,8 +33,30 @@ export class UsersService {
     return this.userRepository.save(user);
   }
 
+  async createTokenForUser(
+    userName: string,
+    identifier: string,
+  ): Promise<AuthToken> {
+    const user = await this.getUserByUsername(userName);
+    let accessToken = '';
+    for (let i = 0; i < 100; i++) {
+      try {
+        const randomString = crypt.randomBytes(64).toString();
```

Comment:
This will just convert random bytes to a string, including invalid character points and control characters. Even if the `hashPassword` method is resistent to this, we should keep in mind that an access token probably only should contain printable characters.

My suggestion is to use either the hexadecimal representation of these bytes or the base64 encoded version of these bytes.

For a comparison see my example snippet below:
![image](https://user-images.githubusercontent.com/52606896/104850502-c629fa80-58ef-11eb-8ba2-41a7141b719a.png)

Reply:
done

Reply:
Just in case this isn't done yet (didn't check the new code) please use base64url, not regular base64 as those /-characters can be beasts.